src/components/owned/OwnerInfo.tsx

```diff
@@ -29,82 +28,73 @@ const walletBox = classnames(
-          >
-            <AccentText bold color="text-secondary">
-              <ContractName
-                hyphens
-                address={badge.address}
-                network={Network.Goerli}
-              />
-            </AccentText>
-          </ExternalLink>
-        </HeaderText>
-      </>
+function replaceKeywordsInString(
+  template: string,
+  keyword: string,
+  element: JSXInternal.Element
+) {
+  return template
+    .split(keyword)
+    .reduce(
+      (chain, part, index) =>
+        index === 0 ? chain.concat(part) : chain.concat(element).concat(part),
+      [] as (JSXInternal.Element | string)[]
     )
-  }
+}
 
-  return (
-    <>
-      <HeaderText extraLeading>
-        <span className={display('md:block')}>This wallet owns </span>a{' '}
-        <ExternalLink
-          url={getEtherscanAddressUrl(badge.address, Network.Goerli)}
-        >
-          <AccentText bold color="text-secondary">
-            <ContractName
-              hyphens
-              address={badge.address}
-              network={Network.Goerli}
-            />
-          </AccentText>
-        </ExternalLink>
-      </HeaderText>
-    </>
+function BadgeTitle({
+  template,
+  derivative,
+}: {
+  template: string
+  derivative: string
+}) {
+  if (!template) return null
+
+  const derivativeLink = (
+    <ExternalLink url={getEtherscanAddressUrl(derivative, Network.Goerli)}>
+      <AccentText bold color="text-secondary">
+        <ContractName hyphens address={derivative} network={Network.Goerli} />
+      </AccentText>
+    </ExternalLink>
+  )
+
+  const title = replaceKeywordsInString(
+    template,
+    '{derivative}',
+    derivativeLink
   )
+
+  return <HeaderText extraLeading>{title}</HeaderText>
 }
 
-function BadgeContent({ badge }: { badge: BaseBadgeContract }) {
-  if (badge instanceof EmailBadgeContract) {
-    return (
-      <BodyText>
-        This is a zkNFT derivative of an email. It means this person has been
-        verified own a ‘
-        <AccentText color="text-secondary">{badge.domain}</AccentText>‘ email.
-        <ShareCTAButtons />
-      </BodyText>
-    )
-  }
+function BadgeContent({
+  template,
+  original,
+  network,
+}: {
+  template: string
+  original: string
+  network: Network
+}) {
+  if (!template) return null
 
-  if (badge instanceof ERC721BadgeContract) {
-    return (
-      <BodyText>
-        This is a zkNFT derivative. It means this person has been verified to
-        own at least one ‘
-        <ExternalLink
-          url={getEtherscanAddressUrl(badge.originalERC721, badge.network)}
-        >
-          <AccentText color="text-secondary">
-            <ContractName
-              address={badge.originalERC721}
-              network={badge.network}
-            />
-          </AccentText>
-        </ExternalLink>
-        ‘ {badge.network[0].toUpperCase() + badge.network.slice(1)} NFT.
-        <ShareCTAButtons />
-      </BodyText>
-    )
-  }
+  const originalLink = (
```

Comment:
This component looks identically like `**derivativeLink**`. Could we do smt with it?